src/mito2/src/compaction/compactor.rs

```diff
@@ -0,0 +1,422 @@
+    );
+
+    let object_store = {
+        let name = &region_options.storage;
+        if let Some(name) = name {
+            object_store_manager
+                .find(name)
+                .context(ObjectStoreNotFoundSnafu {
+                    object_store: name.to_string(),
+                })?
+        } else {
+            object_store_manager.default_object_store()
+        }
+    };
+
+    let mut mito_config = MitoConfig::default();
+    mito_config.sanitize(data_home)?;
+
+    let access_layer = {
+        let intermediate_manager =
+            IntermediateManager::init_fs(mito_config.inverted_index.intermediate_path.clone())
+                .await?;
+
+        Arc::new(AccessLayer::new(
+            region_dir.to_string(),
+            object_store.clone(),
+            intermediate_manager,
+        ))
+    };
+
+    let manifest_manager = {
+        let region_manifest_options = RegionManifestOptions {
+            manifest_dir: join_dir(&region_dir, "manifest"),
+            object_store: object_store.clone(),
+            compress_type: manifest_compress_type(mito_config.compress_manifest),
+            checkpoint_distance: mito_config.manifest_checkpoint_distance,
+        };
+
+        RegionManifestManager::open(region_manifest_options, Default::default())
+            .await?
+            .context(EmptyRegionDirSnafu {
+                region_id: req.region_id,
+                region_dir,
+            })?
+    };
+
+    let manifest = manifest_manager.manifest();
+    let region_metadata = manifest.metadata.clone();
+    let manifest_ctx = Arc::new(ManifestContext::new(manifest_manager, Writable));
+
+    let file_purger = {
+        let purge_scheduler = Arc::new(LocalScheduler::new(mito_config.max_background_jobs));
+        Arc::new(LocalFilePurger::new(
+            purge_scheduler.clone(),
+            access_layer.clone(),
+            None,
+        ))
+    };
+
+    let version_control = {
+        let memtable_builder = MemtableBuilderProvider::new(None, Arc::new(mito_config.clone()))
+            .builder_for_options(
+                region_options.memtable.as_ref(),
+                !region_options.append_mode,
+            );
+
+        // Initial memtable id is 0.
+        let mutable = Arc::new(TimePartitions::new(
+            region_metadata.clone(),
+            memtable_builder.clone(),
+            0,
+            region_options.compaction.time_window(),
+        ));
+
+        let version = VersionBuilder::new(region_metadata.clone(), mutable)
+            .add_files(file_purger.clone(), manifest.files.values().cloned())
+            .flushed_entry_id(manifest.flushed_entry_id)
+            .flushed_sequence(manifest.flushed_sequence)
+            .truncated_entry_id(manifest.truncated_entry_id)
+            .compaction_time_window(manifest.compaction_time_window)
+            .options(region_options.clone())
+            .build();
+        Arc::new(VersionControl::new(version))
+    };
```

Comment:
It should be possible to share the same code with `RegionOpener`. Maybe a method `create_version_control()`.

Is it possible to provide a method like `open_for_compact()` in the `RegionOpener`?

Reply:
Make sense. Let me provide one.

Reply:
I tried it again, but it seems strange in `RegionOpener`: it's difficult to write an `open_for_compact()` in `RegionOpener{}` because of the arguments dependency. Maybe the `open_compaction_region()` will be the simplest solution.